src/haddock/modules/topology/topocg/aa2cg.py

```diff
@@ -410,45 +431,49 @@ def determine_ss(structure, skipss, pdbf_path):
     Returns:
 
     """
+
     # calculate SS
     for model in structure:
+        tmp_file_name = f"../{output_path}/{pdbf_path.split('/')[-1][:-4]}_tmp.pdb"
+        create_file_with_cryst(pdbf_path, tmp_file_name)
```

Comment:
Here it's best to also use python to generate the temporary file:

```python
tmp_file_name = tempfile.NamedTemporaryFile(delete=False, suffix=".pdb").name
```

This will make sure the temporary file always exists and it's always valid, regardless of paths and such

Reply:
I will add this too :)

Reply:
you can actually create the temporary file inside the function, think it makes it more clean

```python
tmp_file_name = create_file_with_cryst(pdbf_path)
```

Reply:
I actually need the name of the temporary file in the next steps, but I will think of a solution